adaptations/Roland_MKS50.py

```
@@ -0,0 +1,636 @@
+        raise Exception("MKS-50 calculateFingerprint expects one sysex message")
+
+    data = messages[0]
+    if _is_tone_apr(data):
+        # Ignore channel, name and wrapper bytes for better duplicate detection.
+        payload = data[7:43]
+        return hashlib.md5(bytearray(payload)).hexdigest()
+
+    if _is_tone_bld(data):
+        packed = _extract_single_packed_patch(data, 0, 9)
+        apr_data = _apply_tone_mapping(packed)
+        return hashlib.md5(bytearray(apr_data)).hexdigest()
```

Ruff (0.15.10)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)